TTS/engine_wrapper.py

```diff
@@ -29,105 +30,113 @@ class TTSEngine:
+        # so that it won't appear in the final video
         try:
-            Path(f"{self.path}/posttext.mp3").unlink()
+            Path(f'{self.path}/posttext.mp3').unlink()
         except OSError:
             pass
 
-        print_step("Saving Text to MP3 files...")
-
-        self.call_tts("title", self.reddit_object["thread_title"])
-        if (
-            self.reddit_object["thread_post"] != ""
-            and settings.config["settings"]["storymode"] == True
-        ):
-            self.call_tts("posttext", self.reddit_object["thread_post"])
-
-        idx = None
-        for idx, comment in track(enumerate(self.reddit_object["comments"]), "Saving..."):
-            # ! Stop creating mp3 files if the length is greater than max length.
-            if self.length > self.max_length:
-                break
-            if (
-                len(comment["comment_body"]) > self.tts_module.max_chars
-            ):  # Split the comment if it is too long
-                self.split_post(comment["comment_body"], idx)  # Split the comment
-            else:  # If the comment is not too long, just call the tts engine
-                self.call_tts(f"{idx}", comment["comment_body"])
-
-        print_substep("Saved Text to MP3 files successfully.", style="bold green")
-        return self.length, idx
-
-    def split_post(self, text: str, idx: int):
-        split_files = []
-        split_text = [
-            x.group().strip()
-            for x in re.finditer(
-                r" *(((.|\n){0," + str(self.tts_module.max_chars) + "})(\.|.$))", text
-            )
-        ]
-        offset = 0
-        for idy, text_cut in enumerate(split_text):
-            # print(f"{idx}-{idy}: {text_cut}\n")
-            if not text_cut or text_cut.isspace():
-                offset += 1
-                continue
-
-            self.call_tts(f"{idx}-{idy - offset}.part", text_cut)
-            split_files.append(AudioFileClip(f"{self.path}/{idx}-{idy - offset}.part.mp3"))
-
-        CompositeAudioClip([concatenate_audioclips(split_files)]).write_audiofile(
-            f"{self.path}/{idx}.mp3", fps=44100, verbose=False, logger=None
-        )
+        print_step('Saving Text to MP3 files...')
 
-        for i in split_files:
-            name = i.filename
-            i.close()
-            Path(name).unlink()
+        self.call_tts('title', self.reddit_object['thread_title'])
 
-        # for i in range(0, idy + 1):
-        # print(f"Cleaning up {self.path}/{idx}-{i}.part.mp3")
+        if self.reddit_object['thread_post'] and settings.config['settings']['storymode']:
+            self.call_tts('posttext', self.reddit_object['thread_post'])
 
-        # Path(f"{self.path}/{idx}-{i}.part.mp3").unlink()
+        sync_tasks_primary = [
+            self.call_tts(str(idx), comment['comment_body'])
+            for idx, comment in track(enumerate(self.reddit_object['comments']), description='Saving...')
```

Comment:
track(enumerate()) are in the wrong order here

Reply:
there is no difference, is there?

it work either way

Reply:
@JasonLovesDoggo 
can you elaborate? 
both of them return iterator, track print stuff in dunder next on top of iter while enumerate adds another value in next and returns 2 iters

am I blind or what?

Reply:
I don't have time for this right now as I'm about to go camping ask cordless coder

Reply:
@CordlessCoder 
can you elaborate then?

Reply:
> @CordlessCoder can you elaborate then?

enumerate doesn't correctly pass the total length of the enumerator in the correct format for track to parse. It just doesn't calculate progress when you enumerate before tracking.